python/nammu/controller/AtfAreaController.py

```diff
@@ -219,6 +227,82 @@ def pad_bottom_viewport_caret(self, bottom_left_char, text):
 
         return bottom_left_char
 
+    def update_error_lines(self, caret_line, no_of_lines, flag):
+        '''
+        Given a caret line number, a number of lines and a flag indicating
+        whether the error lines need incremented ('insert') or decremented
+        ('remove'). Update the line numbers of the keys in the dictionary
+        self.validation_errors so that error highlighting follows broken lines
+        during editing.
+        '''
+
+        # If the supplied edit does not add or remove any lines, do nothing
+        if no_of_lines < 1:
+            return
+
+        error_lines = self.validation_errors.keys()
+
+        # For legacy reasons the keys are strings, but we need ints
+        e_lines_int = [int(a) for a in error_lines]
+
+        # We only care about edits hapenning above error lines
+        if caret_line > max(e_lines_int):
+            return
```

Comment:
I think it is more efficient to keep these 2 if statements split. If we want to merge the 2 statements, we need to do the list comprehension to generate `e_lines_int` prior to the if statement. This results in processing the comprehension every time there is an edit, rather than only when there is an insertion or removal of a line.